### PR TITLE
chore(java indexer): Store absolute path

### PR DESCRIPTION
**Pull request by salguarnieri**
Store the absolute path for the temp directory so we don't have to compute it for every getPath call

### CODE REVIEW

File: kythe/java/com/google/devtools/kythe/platform/java/filemanager/CompilationUnitPathFileManager.java [hunk 1]
```diff
@@ -58,10 +58,11 @@
 
   private final CompilationUnitFileSystem fileSystem;
   private final ImmutableSet<String> defaultPlatformClassPath;
-  // The path given to us that we are allowed to write in.
+  // The path given to us that we are allowed to write in. This should be stored as an absolute
```
Comment (jaysachs) [inline kythe/java/com/google/devtools/kythe/platform/java/filemanager/CompilationUnitPathFileManager.java]: s/should/will/ ?

File: kythe/java/com/google/devtools/kythe/platform/java/filemanager/CompilationUnitPathFileManager.java [hunk 2]
```diff
@@ -58,10 +58,11 @@
...
+  // path.
   private final @Nullable Path temporaryDirectoryPrefix;
   // A temporary directory inside of temporaryDirectoryPrefix that we will use and delete when the
-  // close method is called.
+  // close method is called. This should be stored as an absolute path.
```
Comment (jaysachs) [inline kythe/java/com/google/devtools/kythe/platform/java/filemanager/CompilationUnitPathFileManager.java]: s/should/will/ ?

File: kythe/java/com/google/devtools/kythe/platform/java/filemanager/CompilationUnitPathFileManager.java (hunk 1 above)
Comment (salguarnieri) [reply kythe/java/com/google/devtools/kythe/platform/java/filemanager/CompilationUnitPathFileManager.java]: 👍

File: kythe/java/com/google/devtools/kythe/platform/java/filemanager/CompilationUnitPathFileManager.java (hunk 2 above)
Comment (salguarnieri) [reply kythe/java/com/google/devtools/kythe/platform/java/filemanager/CompilationUnitPathFileManager.java]: 👍

File: kythe/java/com/google/devtools/kythe/platform/java/filemanager/CompilationUnitPathFileManager.java [hunk 3]
```diff
@@ -58,10 +58,11 @@
 
   private final CompilationUnitFileSystem fileSystem;
   private final ImmutableSet<String> defaultPlatformClassPath;
-  // The path given to us that we are allowed to write in.
+  // The path given to us that we are allowed to write in. This will be stored as an absolute
+  // path.
   private final @Nullable Path temporaryDirectoryPrefix;
   // A temporary directory inside of temporaryDirectoryPrefix that we will use and delete when the
-  // close method is called.
+  // close method is called. This should be will as an absolute path.
```
Comment (jaysachs) [inline kythe/java/com/google/devtools/kythe/platform/java/filemanager/CompilationUnitPathFileManager.java]: will be stored

Comment (salguarnieri) [reply kythe/java/com/google/devtools/kythe/platform/java/filemanager/CompilationUnitPathFileManager.java]: https://github.com/kythe/kythe/pull/4224

**Pull request merged**